cli/permissions.rs

```diff
@@ -170,51 +180,108 @@ impl DenoPermissions {
   pub fn check_read(&self, filename: &str) -> DenoResult<()> {
     match self.allow_read.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("read access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_read.update_with_prompt_result(&v);
-          v.check()?;
+      state => {
+        if check_path_white_list(filename, &self.read_whitelist) {
           Ok(())
+        } else {
+          match state {
+            // This shouldn't be possible but I guess rust doesn't realize.
+            PermissionAccessorState::Allow => unreachable!(),
+            PermissionAccessorState::Ask => match self.try_permissions_prompt(
+              &format!("read access to \"{}\"", filename),
+            ) {
+              Err(e) => Err(e),
+              Ok(v) => {
+                self.allow_read.update_with_prompt_result(&v);
+                v.check()?;
+                Ok(())
+              }
+            },
+            PermissionAccessorState::Deny => Err(permission_denied()),
+          }
         }
-      },
-      PermissionAccessorState::Deny => Err(permission_denied()),
+      }
     }
   }
 
   pub fn check_write(&self, filename: &str) -> DenoResult<()> {
     match self.allow_write.get_state() {
       PermissionAccessorState::Allow => Ok(()),
-      PermissionAccessorState::Ask => match self
-        .try_permissions_prompt(&format!("write access to \"{}\"", filename))
-      {
-        Err(e) => Err(e),
-        Ok(v) => {
-          self.allow_write.update_with_prompt_result(&v);
-          v.check()?;
+      state => {
+        if check_path_white_list(filename, &self.write_whitelist) {
           Ok(())
+        } else {
+          match state {
+            // This shouldn't be possible but I guess rust doesn't realize.
+            PermissionAccessorState::Allow => unreachable!(),
+            PermissionAccessorState::Ask => match self.try_permissions_prompt(
+              &format!("write access to \"{}\"", filename),
+            ) {
+              Err(e) => Err(e),
+              Ok(v) => {
+                self.allow_write.update_with_prompt_result(&v);
+                v.check()?;
+                Ok(())
+              }
+            },
+            PermissionAccessorState::Deny => Err(permission_denied()),
+          }
         }
-      },
-      PermissionAccessorState::Deny => Err(permission_denied()),
+      }
     }
   }
 
-  pub fn check_net(&self, domain_name: &str) -> DenoResult<()> {
+  pub fn check_net(&self, url_str: &str) -> DenoResult<()> {
```

Comment:
I don't think check_net should take a URL. You can add a new method which checks a URL, but, for example, `Deno.dial()` does not use URLs.

Reply:
I'm going to refactor this to `origin`(host/domain + optional port) for now. I still think we may want to move to full urls at some point for more parsing consistency, but I'll save that for another PR. For `dial()` it wouldn't be that hard to just add the protocol on for a full url I.E. `tcp://localhost:1234`.

Reply:
One reason url might be useful is restricting libs, e.g. `deno.land/std` or `deno.land/x/oak` (rather than everything in the registry `deno.land/x/`).

Reply:
Thinking about it host+port is probably sufficient for requests, I was confusing this with imports which is less relevant here.